builder/config.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2023 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package builder
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kiegroup/container-builder/util/log"
+	operatorapi "github.com/kiegroup/kogito-serverless-operator/api/v1alpha08"
+)
+
+const (
+	envVarPodNamespaceName = "POD_NAMESPACE"
+	// ConfigMapName is the default name for the Builder ConfigMap name
+	ConfigMapName                       = "kogito-serverless-operator-builder-config"
+	configKeyDefaultExtension           = "DEFAULT_WORKFLOW_EXTENSION"
+	configKeyDefaultBuilderResourceName = "DEFAULT_BUILDER_RESOURCE_NAME"
+	configKeyBuildNamespace             = "build-namespace"
+	configKeyRegistrySecret             = "registry-secret"
+	configKeyRegistryAddress            = "registry-address"
+)
+
+func NewCustomConfig(platform operatorapi.KogitoServerlessPlatform) (map[string]string, error) {
+	customConfig := make(map[string]string)
+	if platform.Namespace == "" {
+		return nil, errors.New(fmt.Sprintf("Unable to retrieve the namespace from platform %s", platform.Name))
```

Comment:
ins't better to use fmt.Errorf?

Reply:
This is a move from an old file, I'll change.